### PR TITLE
Same optimizer settings with ``settings.optimizer.enabled`` missing and ``false``

### LINKED ISSUE #11521: Different optimizer settings in Standard JSON when `optimizer` dict is empty
## Description
In Standard JSON an empty `optimizer` dict has a different effect than one with explicitly enabled or disabled optimization:
- `"settings": {"optimizer": {"enabled": false}}` uses [`OptimiserSettings::minimal()`](https://github.com/ethereum/solidity/blob/v0.8.5/libsolidity/interface/OptimiserSettings.h#L68-L74).
- `"settings": {"optimizer": {"enabled": true}}`  uses [`OptimiserSettings::standard()`](https://github.com/ethereum/solidity/blob/v0.8.5/libsolidity/interface/OptimiserSettings.h#L76-L90).
- `"settings": {"optimizer": {}}` uses [`OptimiserSettings::none()`](https://github.com/ethereum/solidity/blob/v0.8.5/libsolidity/interface/OptimiserSettings.h#L63-L66).

This might be intentional - using `none()` when the dict is empty makes sense in a way. The problem is that this is not at all obvious to the user. One might reasonably assume that not including `enabled` at all would be equivalent to `"enabled": false`, especially given that non-JSON interface does not even provide any third choice.

This has already resulted in a bug in solc-js (https://github.com/ethereum/solc-js/issues/527) where the lack of `--optimize` flag has a different result than in solc.

I think that minimal settings should be used when `enabled` is missing. Alternatively, we could replace the boolean with an explicit choice of optimization level if we want it to be still possible to select `OptimiserSettings::none()`.

For reference, here's the code in `StandardCompiler` that handles the flag: https://github.com/ethereum/solidity/blob/a4f2e591fe55900f86ab72fad37d03af7dfa638e/libsolidity/interface/StandardCompiler.cpp#L561-L569

### PR DESCRIPTION
**Pull request by cameel**
Fixes #11521.

### CODE REVIEW

**Comment by cameel** [comment]
I wanted to include a command-line test but it would only make sense if the output was metadata so that we can see the flag. And including metadata would include compiler version so no test for this unfortunately.

**Review by hrkrshnn — DISMISSED** [review]

**Review by chriseth — DISMISSED** [review]

File: Changelog.md [hunk 1]
```diff
@@ -9,6 +9,7 @@ Compiler Features:
 Bugfixes:
  * Control Flow Graph: Fix incorrectly reported unreachable code.
  * Solc-Js: Use the same optimizer settings as ``solc`` when the ``--optimize`` flag is not specified.
+ * Standard JSON: Fix missing ``settings.optimizer.enabled`` key not being equivalent to setting it to ``false``.
```
Comment (chriseth) [inline Changelog.md]: Actually can you write a bit more about what it was doing? I mean just that people don't think that the optimizer was fully enabled.

Comment (cameel) [reply Changelog.md]: Sure.

Comment (cameel) [reply Changelog.md]: Updated with the text I posted on the chat.

**Review by chriseth — APPROVED** [review]

**Pull request merged**